src/include/type/expression/case_expression.h

```diff
@@ -0,0 +1,128 @@
+#pragma once
+
+#include <memory>
+#include <utility>
+#include <vector>
+#include "type/expression/abstract_expression.h"
+
+namespace terrier {
+namespace type {
+namespace expression {
+
+/**
+ * Represents a logical case expression.
+ */
+class CaseExpression : public AbstractExpression {
+ public:
+  /**
+   * A unique pointer to an abstract expression.
+   */
+  using AbsExprPtr = std::unique_ptr<AbstractExpression>;
```

Comment:
A lot of files in this folder use std::unique_ptr<AbstractExpression>. Seems odds to me to only have it in this one file. I'd remove this or add it to expression_defs.h and use it consistently in all of these other files.